subxt/src/backend/legacy/mod.rs

```diff
@@ -84,183 +85,249 @@ impl<T: Config + Send + Sync + 'static> Backend<T> for LegacyBackend<T> {
+                .filter_map(|r| future::ready(r.transpose()));
 
-        let s = stream::iter(iter)
-            // Resolve the future
-            .then(|fut| fut)
-            // Filter any Options out (ie if we didn't find a value at some key we return nothing for it).
-            .filter_map(|r| future::ready(r.transpose()));
+            Ok(StreamOf(Box::pin(s)))
+        }
 
-        Ok(StreamOf(Box::pin(s)))
+        retry(|| fut(keys.clone(), at, self.methods.clone())).await
     }
 
     async fn storage_fetch_descendant_keys(
         &self,
         key: Vec<u8>,
         at: T::Hash,
     ) -> Result<StreamOfResults<Vec<u8>>, Error> {
-        let keys = StorageFetchDescendantKeysStream {
-            at,
-            key,
-            storage_page_size: self.storage_page_size,
-            methods: self.methods.clone(),
-            done: Default::default(),
-            keys_fut: Default::default(),
-            pagination_start_key: None,
-        };
+        retry(|| async {
+            let keys = StorageFetchDescendantKeysStream {
+                at,
+                key: key.clone(),
+                storage_page_size: self.storage_page_size,
+                methods: self.methods.clone(),
+                done: Default::default(),
+                keys_fut: Default::default(),
+                pagination_start_key: None,
+            };
 
-        let keys = keys.flat_map(|keys| {
-            match keys {
-                Err(e) => {
-                    // If there's an error, return that next:
-                    Either::Left(stream::iter(std::iter::once(Err(e))))
-                }
-                Ok(keys) => {
-                    // Or, stream each "ok" value:
-                    Either::Right(stream::iter(keys.into_iter().map(Ok)))
+            let keys = keys.flat_map(|keys| {
+                match keys {
+                    Err(e) => {
+                        // If there's an error, return that next:
+                        Either::Left(stream::iter(std::iter::once(Err(e))))
+                    }
+                    Ok(keys) => {
+                        // Or, stream each "ok" value:
+                        Either::Right(stream::iter(keys.into_iter().map(Ok)))
+                    }
                 }
-            }
-        });
+            });
 
-        Ok(StreamOf(Box::pin(keys)))
+            Ok(StreamOf(Box::pin(keys)))
+        })
+        .await
     }
 
     async fn storage_fetch_descendant_values(
         &self,
         key: Vec<u8>,
         at: T::Hash,
     ) -> Result<StreamOfResults<StorageResponse>, Error> {
-        let keys_stream = StorageFetchDescendantKeysStream {
-            at,
-            key,
-            storage_page_size: self.storage_page_size,
-            methods: self.methods.clone(),
-            done: Default::default(),
-            keys_fut: Default::default(),
-            pagination_start_key: None,
-        };
+        retry(|| async {
```

Comment:
Same; is this `retry` doing anything? No `.await`s so no chance of a `DisconnectedWillReconnect` error during the async block?